src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the humbug/php-scoper package.
+ *
+ * Copyright (c) 2017 Théo FIDRY <theo.fidry@gmail.com>,
+ *                    Pádraic Brady <padraic.brady@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Humbug\PhpScoper\Console\Command;
+
+use Humbug\PhpScoper\Logger\UpdateConsoleLogger;
+use Humbug\SelfUpdate\Updater;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+final class SelfUpdateCommand extends Command
+{
+    /** @internal */
+    const REMOTE_FILENAME = 'php-scoper.phar';
+    /** @internal */
+    const STABILITY_STABLE = 'stable';
+    /** @internal */
+    const PACKAGIST_PACKAGE_NAME = 'humbug/php-scoper';
+    /** @internal */
+    const ROLLBACK_OPT = 'rollback';
+    /** @internal */
+    const CHECK_OPT = 'check';
+
+    /**
+     * @var Updater
+     */
+    private $updater;
+
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    /**
+     * @var UpdateConsoleLogger
+     */
+    private $logger;
+
+    /**
+     * @param Updater $updater
+     */
+    public function __construct(Updater $updater)
+    {
+        parent::__construct();
+
+        $this->version = $this->getApplication()->getVersion();
+        $this->updater = $updater;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription(sprintf(
+                    'Update %s to most recent stable build.',
+                    $this->getLocalPharName()
+            ))
+            ->addOption(
+                self::ROLLBACK_OPT,
+                'r',
+                InputOption::VALUE_NONE,
+                'Rollback to previous version of PHP-Scoper if available on filesystem.'
+            )
+            ->addOption(
+                self::CHECK_OPT,
+                'c',
+                InputOption::VALUE_NONE,
+                'Checks whether an update is available.'
+            )
+        ;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->logger = new UpdateConsoleLogger(
+            new SymfonyStyle($input, $output)
+        );
+
+        $this->output = $output;
+
+        if ($input->getOption('rollback')) {
+            $this->rollback();
+
+            return;
+        }
+
+        if ($input->getOption('check')) {
+            $this->printAvailableUpdates();
+
+            return;
```

Comment:
Null is converted to 0 status automatically isn't it? No harm being explicit though.